source/templates.tex

```diff
@@ -2800,24 +2800,25 @@
 \end{example}
 
 \pnum
-The instantiation of a pack expansion
-that is neither a \tcode{sizeof...} expression
-nor a \grammarterm{fold-expression}
-produces a
-list of elements
-$\mathtt{E}_1,$ $\mathtt{E}_2,$ $\cdots,$ $\mathtt{E}_N$,
+The instantiation of a pack expansion considers
+items $\tcode{E}_1, \tcode{E}_2, \dotsc, \tcode{E}_N$,
 where
-$N$ is the number of elements in the pack expansion parameters. Each
-$\mathtt{E}_i$ is generated by instantiating the pattern and
+$N$ is the number of elements in the pack expansion parameters.
+Each $\tcode{E}_i$ is generated by instantiating the pattern and
 replacing each pack expansion parameter with its $i^\text{th}$ element.
 Such an element, in the context of the instantiation, is interpreted as
 follows:
 \begin{itemize}
 \item
-if the pack is a template parameter pack, the element is a template
-parameter\iref{temp.param} of the corresponding kind (type or
-non-type) designating the $i^\text{th}$
-corresponding type or value template argument;
+if the pack is a template parameter pack, the element is
+an \grammarterm{id-expression}
+(for a non-type template parameter pack),
+a \grammarterm{typedef-name}
+(for a type template parameter pack declared without \tcode{template}), or
+a \grammarterm{template-name}
+(for a type template parameter pack declared with \tcode{template})
+if the pack was introduced by a pack expansion,
```

Comment:
I still think "if the pack was introduced by a pack expansion" should be removed for bullet 8.1. For example, `template<int...N> struct B{ B(){fun(N...)}};`, Isn't that each element instantiated in `N...` does designate the corresponding template argument? However, `int... N` is just a template parameter pack declaration instead of a pack expansion. If add the condition seems that the latter clause is not suitable for this case.